qa/integration-tests/src/test/java/io/zeebe/broker/it/client/command/ActivateJobsTest.java

```diff
@@ -0,0 +1,155 @@
+
+    // when
+    final ActivateJobsResponse response =
+        CLIENT_RULE
+            .getClient()
+            .newActivateJobsCommand()
+            .jobType(jobType)
+            .maxJobsToActivate(activateJobs)
+            .send()
+            .join();
+
+    // then
+    assertThat(response.getJobs()).hasSize(activateJobs);
+  }
+
+  @Test
+  public void shouldActivateJobsIfBatchIsTruncated() {
+    // given
+    final int availableJobs = 10;
+
+    final ByteValue maxMessageSize = BROKER_RULE.getBrokerCfg().getNetwork().getMaxMessageSize();
+    final var largeVariableValue = "x".repeat((int) maxMessageSize.toBytes() / 4);
+    final String variablesJson = String.format("{\"variablesJson\":\"%s\"}", largeVariableValue);
+
+    CLIENT_RULE.createJobs(jobType, b -> {}, variablesJson, availableJobs);
+
+    // when
+    final var response =
+        CLIENT_RULE
+            .getClient()
+            .newActivateJobsCommand()
+            .jobType(jobType)
+            .maxJobsToActivate(availableJobs)
+            .send()
+            .join();
+
+    // then
+    assertThat(response.getJobs()).hasSize(availableJobs);
+  }
+
+  @Test
+  public void shouldWaitUntilJobsAvailable() {
+    // given
+    final int expectedJobsCount = 1;
+
+    final ZeebeFuture<ActivateJobsResponse> responseFuture =
+        CLIENT_RULE
+            .getClient()
+            .newActivateJobsCommand()
+            .jobType(jobType)
+            .maxJobsToActivate(expectedJobsCount)
+            .send();
+
+    // when
+    CLIENT_RULE.createSingleJob(jobType);
+
+    // then
+    final ActivateJobsResponse response = responseFuture.join();
+    assertThat(response.getJobs()).hasSize(expectedJobsCount);
+  }
+
+  @Test
+  public void shouldActivatedJobForOpenRequest() throws InterruptedException {
+    // given
+    sendActivateRequestsAndClose(jobType, 3);
+
+    final var activateJobsResponse =
+        CLIENT_RULE
+            .getClient()
+            .newActivateJobsCommand()
+            .jobType(jobType)
+            .maxJobsToActivate(5)
+            .workerName("open")
+            .send();
+
+    sendActivateRequestsAndClose(jobType, 3);
+
+    // when
+    CLIENT_RULE.createSingleJob(jobType);
+
+    // then
+    final var jobs = activateJobsResponse.join().getJobs();
+
+    assertThat(jobs).hasSize(1).extracting(ActivatedJob::getWorker).contains("open");
+  }
+
+  private void sendActivateRequestsAndClose(String jobType, int count) throws InterruptedException {
+    for (int i = 0; i < count; i++) {
+      final ZeebeClient client =
+          ZeebeClient.newClientBuilder()
+              .brokerContactPoint(BROKER_RULE.getGatewayAddress().toString())
+              .usePlaintext()
+              .build();
+
+      client
+          .newActivateJobsCommand()
+          .jobType(jobType)
+          .maxJobsToActivate(5)
+          .workerName("closed-" + i)
+          .send();
+
+      Thread.sleep(100);
```

Comment:
We could use the recording exporter here to check when the activate requests comes in instead of sleeping?

Reply:
It would be nice to get rid of the `sleep()` but I think it doesn't work with the RecordingExporter because the requests are blocked at the gateway after the first one because of the long-polling.